$NetBSD: patch-CMakeLists.txt,v 1.3 2020/07/04 14:11:41 schmonz Exp $

Support PKGMANDIR.
Support explicit_memset on NetBSD.
Skip building API docs with Doxygen.

--- CMakeLists.txt.orig	2020-01-06 19:19:59.000000000 +0000
+++ CMakeLists.txt
@@ -7,7 +7,7 @@ project(Qsmtp
 
 option(CHECK_MEMORY "Add memory access checks" OFF)
 
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
+set (CMAKE_MODULE_PATH "/data/scratch/mail/qsmtp/work/.buildlink/cmake-Modules" ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
 
 set(CMAKE_C_STANDARD 99)
 set(CMAKE_C_STANDARD_REQUIRED On)
@@ -95,7 +95,8 @@ endif ()
 
 set(CMAKE_REQUIRED_INCLUDES string.h)
 CHECK_FUNCTION_EXISTS(explicit_bzero HAS_EXP_BZERO)
-if (NOT HAS_EXP_BZERO)
+CHECK_FUNCTION_EXISTS(explicit_memset HAS_EXP_MEMSET)
+if (NOT HAS_EXP_BZERO AND NOT HAS_EXP_MEMSET)
 	set(CMAKE_REQUIRED_INCLUDES bsd/string.h)
 	find_library(LIBBSD NAMES bsd)
 	if (LIBBSD)
@@ -111,6 +112,9 @@ if (NOT HAS_EXP_BZERO)
 		add_definitions(-DINSECURE_BZERO)
 	endif ()
 endif ()
+if (HAS_EXP_MEMSET)
+	add_definitions(-DHAS_EXP_MEMSET)
+endif ()
 set(CMAKE_REQUIRED_INCLUDES fcntl.h)
 CHECK_FUNCTION_EXISTS(openat HAS_OPENAT)
 unset(CMAKE_REQUIRED_INCLUDES)
@@ -190,19 +194,19 @@ endif ()
 option(BUILD_DOC "Build documentation" ON)
 if (BUILD_DOC)
 	# API documentation
-	find_package(Doxygen REQUIRED)
+	# find_package(Doxygen REQUIRED)
 
-	configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile @ONLY)
+	# configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile @ONLY)
 
-	add_custom_target(docu ALL
-			COMMAND ${DOXYGEN_EXECUTABLE}
-			WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+	#	add_custom_target(docu ALL
+	#		COMMAND ${DOXYGEN_EXECUTABLE}
+	#		WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
 
 	if (NOT DOC_DESTINATION_DIRECTORY)
 		set(DOC_DESTINATION_DIRECTORY "doc/")
 	endif ()
 
-	install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/html DESTINATION ${DOC_DESTINATION_DIRECTORY})
+	# install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/html DESTINATION ${DOC_DESTINATION_DIRECTORY})
 
 	# general documentation
 	install(FILES
@@ -220,8 +224,8 @@ if (BUILD_DOC)
 	install(FILES
 		${CMAKE_CURRENT_BINARY_DIR}/Qremote.8
 		${CMAKE_CURRENT_BINARY_DIR}/Qsmtpd.8
-		DESTINATION man/man8)
+		DESTINATION ${CMAKE_INSTALL_MANDIR}/man8)
 	install(FILES
 		${CMAKE_CURRENT_BINARY_DIR}/filterconf.5
-		DESTINATION man/man5)
+		DESTINATION ${CMAKE_INSTALL_MANDIR}/man5)
 endif ()
